compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -2187,6 +2199,17 @@ private ParserRuleContext getNextRuleForCloseBrace(int nextLookahead) {
                     return ParserRuleContext.STATEMENT;
                 } else if (parentCtx == ParserRuleContext.NAMED_WORKER_DECL) {
                     endContext(); // end named-worker
+                    parentCtx = getParentContext();
+                    if (parentCtx == ParserRuleContext.FORK_STMT) {
+                        nextToken = this.tokenReader.peek(nextLookahead);
+                        switch (nextToken.kind) {
+                            case AT_TOKEN:
+                            case WORKER_KEYWORD:
+                                return ParserRuleContext.NAMED_WORKER_DECL;
+                            default:
+                                return ParserRuleContext.CLOSE_BRACE;
```

Comment:
I feel we should make STATEMENT as the next rule. So that it will proceed even when there are other statements thought its not valid. But the parser will do that validation anyway.

Otherwise the issue here is, if there is some additional token before the`@` or `worker` keyword, then this will try searching for a close brace, which is kind of wrong.

Reply:
yea that makes more sense, will change it to statement.